code/clue.js

```diff
@@ -7,53 +7,310 @@ const mrGreen = {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// FUNCTION FOR THE KILLER
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
+document.getElementById('killerCard').onclick = pickKiller;
+
+function pickKiller() {
   mystery.killer = randomSelector(suspects)
+    document.getElementById('killerCard').style.background = mystery.killer.color
+    document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`
+    document.getElementById('killerAge').innerHTML = `${mystery.killer.age}`
+    document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`
+    document.getElementById('killerImage'); killerImage.src = mystery.killer.image;
+}  
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// FUNCTION FOR THE WEAPON
+
+document.getElementById('weaponCard').onclick = pickWeapon;
+
+function pickWeapon() {
+  mystery.weapon = randomSelector(weapons)
+    document.getElementById('weaponCard').style.background = mystery.weapon.color
+    document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`
+    document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight}`
+    document.getElementById('weaponImage'); weaponImage.src = mystery.weapon.image;
+}  
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+// FUNCTION FOR THE ROOM
+
+document.getElementById('roomCard').onclick = pickRoom;
+
+function pickRoom() {
+  mystery.room = randomSelector(rooms)
+    document.getElementById('roomName').innerHTML = `${mystery.room.name}`
+    document.getElementById('roomCard').style.background = mystery.room.color
+}  
+
+  
+// MYSTERY OBJECT
+
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
+}
+  
+  
+// FUNCTION TO REVEAL THE CRIME
+
+const revealMystery = () => {
+  document.getElementById("mystery").innerHTML= `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room.name} with a ${mystery.weapon.name}.`
+  document.getElementById("clickMe").onclick = revealMystery;
+}
```

Comment:
If we leave one of the cards blank, and press `revealMystery`  we see this in the console
![image](https://user-images.githubusercontent.com/807352/93990310-42ba2b80-fd8b-11ea-81f1-6305c586079d.png)

It would be good to get rid of the error, and possibly even prevent the reveal from doing anything until all cards are clicked. But definitely we don't want errors